crates/bevy_ecs/macros/src/states.rs

```diff
@@ -9,36 +10,90 @@ pub fn derive_states(input: TokenStream) -> TokenStream {
     let error = || {
         syn::Error::new(
             Span::call_site().into(),
-            "derive(States) only supports fieldless enums",
+            "derive(States) only supports enums (additionaly, all of the fields need to implement States too)",
```

Comment:
It would be nice to be able to split apart these error cases, but I won't block on it.

Reply:
I think I intended to write that too, but wondered how to write it in a way that isn't scary to users. 

But since you support it, great!